### ISSUE TITLE
Add tests for add_title, add_units, add_precursors

### ISSUE THREAD

**Issue by bpbond** [JGCRI/gcamdata#32]
Also want to have checks that chunks have provided required documentation on their objects. 
ALSO want to check that their precursors are in fact one of their inputs!
Put this logic in `return_data` or the driver?

```R
  # Check that the chunk has provided required data for all objects
  for(at in c(ATTR_TITLE, ATTR_UNITS, ATTR_COMMENTS, ATTR_PRECURSORS)) {
    for(obj in names(dots)) {
       if(is.null(attr(dots[[obj]], at))) {
         warning("No '", at, "' attached to ", obj)
       }
     }
   }
```

**Comment by bpbond** [comment]
See #27

**Comment by bpbond** [comment]
Implemented in 3a11df8.
